.github/linters/.htmlhintrc

```diff
@@ -0,0 +1,25 @@
+{
+  "alt-require": true,
+  "attr-lowercase": ["viewBox"],
+  "attr-value-double-quotes": true,
+  "attr-value-not-empty": false,
+  "attr-no-duplication": true,
+  "attr-unsafe-chars": true,
+  "doctype-first": false,
+  "doctype-html5": true,
+  "empyty-tag-self-close": true,
+  "id-unique": false,
```

Comment:
Do we not want to enforce ID uniqueness?

Reply:
Yes, ideally but can't due to reuse in some templates as per my PR description above.

For example:

https://github.com/HTTPArchive/almanac.httparchive.org/blob/fdbdbe9feab00b302c73078d77abd1ded1659d4f/src/templates/base/2019/base_chapter.html#L184
and also
https://github.com/HTTPArchive/almanac.httparchive.org/blob/fdbdbe9feab00b302c73078d77abd1ded1659d4f/src/templates/base/2019/base_chapter.html#L194
Only one of these ends up in the template as it's surround in `{% if ... %}/{% else %}/{% endif %}` Jinja2 constructs.

And also loads of instances in ebook templated which is generated.

Would be a bit of refactoring to prevent this, and not 100% sure we should to be honest. Could make the Jinja2 code more complex to write and read.

Ideally it would be nice if HTMLHint allowed us to ignore certain ids but it currently doesn't.

I say we go with what we have at present, as still loads of benefits to the other checks. We could then look to enforce it later, or look at linting the actual HTML produced in the end as you suggested elsewhere.

Reply:
Yes, you are right, in conditional templates it is very possible and only a smart template linter can detect the branch in place. We can enforce this if and when we decide to lint the generated HTML later, so I will approve this PR.

Ideally, we would have added a comment next to this rule, explaining why we chose what we did, but unfortunately JSON does not support comments (unless we are using [JSON5](https://json5.org/))

Reply:
I've seen some linters support a comment disabling certain rules for specific lines of code. Does this linter have a feature like that so we can enable the unique ID rule but turn it off in cases where the IDs are conditionally added?

Reply:
I’ve configured it where possible. Note I allow viewBox to be an exception to the rule that attributes must be all lowercase a few lines above.

However this is rule [doesn’t allow any configuration but on or off](https://htmlhint.com/docs/user-guide/rules/id-unique).

Still. I was pleasantly surprised how few rules I had to turn off so think there’s still a lot of value in this. We’ll look at linting output HTML as part of a nightly build as think we can run other stuff (e.g. Lighthouse, test that every page returns a 200... etc.). I’ve broken the sitemap on previous releases when changing python so think that would be a good test script to be able to run offline on demand (e.g. when doing a release) and nightly on `main`.